packages/backend/src/api/users/__tests__/user.service.test.ts

```diff
@@ -6,24 +8,40 @@ jest.mock('../user.repository');
 
 describe('User Service', () => {
   let service: UserService;
+  let hashingService: HashingService;
 
-  beforeAll(() => {
-    const repository = new UserRepository();
-    service = new UserService(repository);
+  beforeAll(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [HashingService],
+    }).compile();
+    hashingService = module.get<HashingService>(HashingService);
+
+    service = new UserService(new UserRepository(), hashingService);
   });
 
   describe('register', () => {
+    let hashPasswordSpy: jest.SpyInstance;
+    const payload = {
+      firstName: 'Khaled',
+      lastName: 'Mohamed',
+      email: 'khaled@skillfuze.com',
+      password: '123456789',
+      confirmPassword: '123456789',
+    };
+
+    beforeEach(() => {
+      hashPasswordSpy = jest.spyOn(hashingService, 'hashPassword');
+    });
+
     it('should register the user successfully', async () => {
-      const payload = {
-        firstName: 'Khaled',
-        lastName: 'Mohamed',
-        email: 'khaled@skillfuze.com',
-        password: '123456789',
-        confirmPassword: '123456789',
-      };
       const user = await service.register(payload);
       expect(user).toBeInstanceOf(User);
     });
+
+    it('should call hashingService.hashPassword', async () => {
+      await service.register(payload);
+      expect(hashPasswordSpy).toBeCalled();
```

Comment:
This should check also that the returned password is hashed (not only to check that it is called)

```suggestion
      const res = await service.register(payload);
      expect(res.password).toBe(hashedPassword);
```